include/libdnf/advisory/advisory_package.hpp

```diff
@@ -0,0 +1,91 @@
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_ADVISORY_ADVISORY_PACKAGE_HPP
+#define LIBDNF_ADVISORY_ADVISORY_PACKAGE_HPP
+
+#include "libdnf/advisory/advisory.hpp"
+
+#include <memory>
+#include <vector>
+
+
+namespace libdnf::advisory {
+
+class AdvisoryPackage {
+public:
+    AdvisoryPackage(const AdvisoryPackage & src);
+    AdvisoryPackage(AdvisoryPackage && src);
+
+    AdvisoryPackage & operator=(const AdvisoryPackage & src);
+    AdvisoryPackage & operator=(AdvisoryPackage && src) noexcept;
+    ~AdvisoryPackage();
+
+    /// Get name of this AdvisoryPackage.
+    ///
+    /// @return Name of this AdvisoryPackage as std::string.
+    std::string get_name() const;
+
+    /// Get version of this AdvisoryPackage.
+    ///
+    /// @return Version of this AdvisoryPackage as std::string.
+    std::string get_version() const;
+
+    /// Get evr of this AdvisoryPackage.
+    ///
+    /// @return Evr of this AdvisoryPackage as std::string.
+    std::string get_evr() const;
+
+    /// Get arch of this AdvisoryPackage.
+    ///
+    /// @return Arch of this AdvisoryPackage as std::string.
+    std::string get_arch() const;
+
+    /// Get AdvisoryId of Advisory this AdvisoryPackage belongs to.
+    ///
+    /// @return AdvisoryId of this AdvisoryPackage.
+    AdvisoryId get_advisory_id() const;
+
+    /// Compare NEVRAs of two AdvisoryPackages
+    ///
+    /// @param first        First AdvisoryPackage to compare.
+    /// @param second       Secondf AdvisoryPackage to compare.
+    /// @return True if first AdvisoryPackage has smaller NEVRA, False otherwise.
+    static bool compareNEVRA(const AdvisoryPackage & first, const AdvisoryPackage & second);
+
+    /// Compare Name and Architecture of AdvisoryPackage and libdnf::rpm::Package
+    ///
+    /// @param adv_pkg      AdvisoryPackage to compare.
+    /// @param pkg          libdnf::rpm::Package to compare.
+    /// @return True if AdvisoryPackage has smaller name or architecture than libdnf::rpm::package, False otherwise.
+    static bool comparePackageNA(const AdvisoryPackage & adv_pkg, const rpm::Package & pkg);
+
+private:
+    friend class AdvisoryCollection;
+    friend class AdvisoryQuery;
+    friend class libdnf::rpm::SolvQuery;
+
+    class Impl;
+    AdvisoryPackage(Impl * private_pkg);
```

Comment:
Not sure if it is the best idea. But the second option would be to use `libdnf::rpm::SolvSack & sack, AdvisoryId advisory, int owner_collection_index, Id name, Id stream, Id version, Id context, Id arch`. But `Id` must be replaced by `int`. And using `int` as `Id` is also horrible. @jrohel What do you think?

Reply:
In case that we will be able to drop p_impl in `AdvisoryPackage, we can improve a performance.

Reply:
Added TODOs with this point to `AdvisoryPackage` and `AdvisoryModule`.